### PR TITLE
Unify spark fs

### PR DESCRIPTION
**Pull request by rabbitonweb**
Refactor code, move common code to fs pacakge object in sparkcore.

After this change adding new spark-connector should be a bit simplier

### CODE REVIEW

**Comment by drostron** [comment]
The CB failure seen on Travis is addressed in #1830.

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala [hunk 1]
```diff
@@ -52,15 +52,17 @@ package object hdfs {
...
-    def forge(master: String, hdfsUriStr: String, rootPath: String): DefinitionError \/ SparkFSConf =
+    def forge(master: String, hdfsUriStr: String, rootPath: String): DefinitionError \/ (SparkConf, SparkFSConf) =
       posixCodec.parseAbsDir(rootPath)
-        .map { prefix =>
-        SparkFSConf(new SparkConf().setMaster(master).setAppName("quasar"), hdfsUriStr, sandboxAbs(prefix))
+        .map { prefix => {
+          val conf = SparkFSConf(new SparkConf().setMaster(master).setAppName("quasar"), hdfsUriStr, sandboxAbs(prefix))
+          (conf.sparkConf, conf)
+        }
       }.fold(error(s"Could not extrat a path from $rootPath"))(_.right[DefinitionError])
```
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: s/extrat/extract/ _(preexisting)_

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/local/package.scala [hunk 2]
```diff
@@ -48,33 +48,31 @@ package object local {
...
+
+     def error(msg: String): DefinitionError \/ (SparkConf, SparkFSConf) =
+        NonEmptyList(msg).left[EnvironmentError].left[(SparkConf, SparkFSConf)]
+
+      def forge(master: String, rootPath: String): DefinitionError \/ (SparkConf, SparkFSConf) =
+        posixCodec.parseAbsDir(rootPath)
+          .map { prefix =>
+          val sc = new SparkConf().setMaster(master)
+          (sc, SparkFSConf(sc, sandboxAbs(prefix)))
+        }.fold(error(s"Could not extrat a path from $rootPath"))(_.right[DefinitionError])
```
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/local/package.scala]: Same

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/packaga.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/*
```
Comment (drostron) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/packaga.scala]: s/packaga.scala/package.scala/

**Comment by rabbitonweb** [comment]
@drostron rename + typeos fixed

**Pull request merged**